code/modules/client/preferences/language.dm

```diff
@@ -33,3 +43,42 @@
 
 /datum/preference/choiced/language/apply_to_human(mob/living/carbon/human/target, value)
 	return
+
+/datum/preference/toggle/language_speakable
+	category = PREFERENCE_CATEGORY_MANUALLY_RENDERED
+	savefile_key = "language_speakable"
+	savefile_identifier = PREFERENCE_CHARACTER
+	default_value = TRUE
+	can_randomize = FALSE
+
+/datum/preference/toggle/language_speakable/is_accessible(datum/preferences/preferences)
+	if(!..())
+		return FALSE
+
+	return /datum/quirk/bilingual::name in preferences.all_quirks
+
+/datum/preference/toggle/language_speakable/apply_to_human(mob/living/carbon/human/target, value)
+	return
+
+/datum/preference/choiced/language_skill
+	category = PREFERENCE_CATEGORY_MANUALLY_RENDERED
+	savefile_key = "language_skill"
+	savefile_identifier = PREFERENCE_CHARACTER
+	can_randomize = FALSE
+
+/datum/preference/choiced/language_skill/create_default_value()
+	return "100%"
+
+/datum/preference/choiced/language_skill/is_accessible(datum/preferences/preferences)
+	if(!..())
+		return FALSE
+	if(preferences.read_preference(/datum/preference/toggle/language_speakable))
+		return FALSE
+
+	return /datum/quirk/bilingual::name in preferences.all_quirks
+
+/datum/preference/choiced/language_skill/init_possible_values()
+	return list("100%", "75%", "50%", "33%", "25%", "10%")
```

Comment:
I feel like this should probably be text descriptors instead of raw percentages, that or making this a slider.

Reply:
A slider or input felt too granular to me, and the preference in the UI has a description.